src/react-cancelable.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Observable } from "rx";
+
+const inTestingEnv = typeof jasmine !== "undefined";
+
+export function Cancelable() {
+	return function decorator(DecoratedComponent) {
+		return class Disposable extends React.Component {
+			disposables = [];
+			render() {
+				return (
+					<DecoratedComponent
+						{...this.props}
+						cancelWhenUnmounted={this.cancelWhenUnmounted}
+						cancelAllSubscriptions={this.cancelAllSubscriptions}
+					/>
+				);
+			}
+			componentWillUnmount() {
+				this.cancelAllSubscriptions();
+			}
+			cancelWhenUnmounted = cancelWhenUnmounted.bind(this);
+			cancelAllSubscriptions = cancelAllSubscriptions.bind(this);
+		};
+	};
+}
+
+export function cancelableModal(El, props = {}) {
+	let el;
+
+	return Observable.create(observer => {
+		el = document.createElement("div");
+		document.body.appendChild(el);
+
+		ReactDOM.render(
+			<El {...props} close={observer.onCompleted} save={observer.onNext} />,
```

Comment:
👍  on this I think this is really cool. Couple questions:

1) Should we pass a prop that allows the modal to onError?
2) Is `save` the best name for onNext? I actually really like changing onCompleted to be `close` because that's intuitive when you're writing a modal. `save` might be the best name for onNext, and maybe `error` for onError? Idk I've gone back and forth between just being  transparent about the observably things that are happening here (keeping the names) or trying modal-ify them (close, save, error).

Reply:
I think passing `onError` down is a good idea. About the names, I don't know. We could just keep the `rx` verbiage and just pass `onCompleted`, `onNext` and `onError` down